crates/bevy_asset/src/lib.rs

```diff
@@ -1856,4 +1887,94 @@ mod tests {
 
     #[derive(Asset, TypePath)]
     pub struct TupleTestAsset(#[dependency] Handle<TestAsset>);
+
+    fn unapproved_path_setup(mode: UnapprovedPathMode) -> (App, GateOpener) {
+        let dir = Dir::default();
+        let a_path = "../a.cool.ron";
+        let a_ron = r#"
+(
+    text: "a",
+    dependencies: [],
+    embedded_dependencies: [],
+    sub_texts: [],
+)"#;
+
+        dir.insert_asset_text(Path::new(a_path), a_ron);
+
+        let mut app = App::new();
+        let (gated_memory_reader, gate_opener) = GatedReader::new(MemoryAssetReader { root: dir });
```

Comment:
None of these tests use the gate. We can just remove the GatedReader and just pass along the MemoryAssetReader directly.